tests/test_rank.py

```diff
@@ -160,3 +160,105 @@ def test_order_label_issues_using_self_confidence_ranking():
     score_idx = sorted(list(zip(scores, indices)), key=lambda y: y[0])  # sort indices by score
     label_issues_indices2 = [z[1] for z in score_idx]
     assert all(label_issues_indices == label_issues_indices2)
+
+
+def test_order_label_issues_using_confidence_weighted_entropy_ranking():
+    label_issues_indices = rank.order_label_issues(
+        label_issues_mask=data["label_errors_mask"],
+        labels=data["labels"],
+        pred_probs=data["pred_probs"],
+        rank_by="confidence_weighted_entropy",
+    )
+    scores = rank.get_confidence_weighted_entropy_for_each_label(data["labels"], data["pred_probs"])
+    indices = np.arange(len(scores))[data["label_errors_mask"]]
+    scores = scores[data["label_errors_mask"]]
+    score_idx = sorted(list(zip(scores, indices)), key=lambda y: y[0])  # sort indices by score
+    label_issues_indices2 = [z[1] for z in score_idx]
+    assert all(label_issues_indices == label_issues_indices2)
+
+
+def test_order_label_issues_using_scoring_func_ranking():
```

Comment:
It seems to me we can easily include the above 2 tests inside of this function as well:
`test_order_label_issues_using_confidence_weighted_entropy_ranking()`, `test_order_label_issues_using_self_confidence_ranking()`.   What I mean is inside this function, you can create a dict object:
```
scoring_methods_dict = {"self_confidence": rank.get_self_confidence_for_each_label, "normalized_margin":  rank.get_normalized_margin_for_each_label, "confidence_weighted_entropy": rank.get_confidence_weighted_entropy_for_each_label}

scoring_methods = scoring_methods_dict.keys()
```
and then add something like this after L197s `assert all(label_issues_indices == label_issues_indices2)
`:

```
score_func = scoring_methods_dict[]
scores = score_func(data["labels"], data["pred_probs"])
 indices = np.arange(len(scores))[data["label_errors_mask"]]  # actually only needs to be called once outside the for loop.
    scores = scores[data["label_errors_mask"]]
    score_idx = sorted(list(zip(scores, indices)), key=lambda y: y[0])  # sort indices by score
    label_issues_indices3 = [z[1] for z in score_idx]
    assert all(label_issues_indices == label_issues_indices3)
```

I prefer doing it that way to reduce the amount of duplicated copy/paste code, and it will be easier to add tests for more scoring functions.

Reply:
Agree, good catch. I combined the tests and also replaced for-loops with @pytest.mark.parametrize. I believe this should take care of the redundancies and make it easier to add tests for more scoring functions in the future.

67612f40b636d2794bfe4b9c3674b19ce4087501
315e956b18f489a90b90a2ca62016b34176997fc
2c83638a255f79b4fea3f6454894bff55c0399b0